names/names.py

```diff
@@ -10,11 +11,36 @@
 names_2 = f.read().split("\n")  # List containing 10000 names
 f.close()
 
+# Original code -- Runtime: 12.05 seconds:
+# duplicates = []
+# for name_1 in names_1:
+#     for name_2 in names_2:
+#         if name_1 == name_2:
+#             duplicates.append(name_1)
+
+# Optimized code using a BST -- Runtime: 0.19s
+bst = None
+
+for name1 in names_1:
+    if bst is None:
```

Comment:
since `bst` will only be None for the first case, you can omit this and just directly do 
`bst = BinarySearchTree(name1)`
and then do the for in loop after
```
for name1 in names_1[1:]:
  bst.insert(name1)  
```

Reply:
But other than that, perfect!